maildir/maildiropen.c

```diff
@@ -122,17 +123,30 @@ int maildir_safeopen_stat(const char *path, int mode, int perm,
 			, perm);
 
 	if (fd < 0)	return (fd);
-	if (fcntl(fd, F_SETFL, (mode & O_APPEND)) || fstat(fd, stat1)
-	    || lstat(path, &stat2))
+	if (fcntl(fd, F_SETFL, (mode & O_APPEND)) || stat1 && fstat(fd, stat1))
```

Comment:
Combining || and && like that often ends up being confusing, without explicit parenthesis, and I'm pretty sure that there are lint checkers that will red-flag this. Let's put in explicit parenthesis here.

Reply:
&& has higher precedence than || but sure, to keep lint and others happy, done.

Btw, I try to follow the existing coding style, but it seems to be varied from place to place, so if perhaps there is a doc somewhere about the correct coding style that I can adhere to that would be great to know about.